unified-installer/astra-unified-installer.sh

```diff
@@ -2436,13 +2540,17 @@ if trident_will_be_installed_or_modified; then
             # Enable ACP if needed (includes ACP upgrade)
             if ! acp_is_enabled; then
                 if config_acp_image_is_custom || prompt_user_yes_no "Would you like to enable ACP?"; then
+                    # create trident-acp secret
+                    kubectl create secret docker-registry "$IMAGE_PULL_SECRET" --docker-username="$ASTRA_ACCOUNT_ID" --docker-password="$ASTRA_API_TOKEN" -n trident --docker-server="$TRIDENT_ACP_IMAGE_REGISTRY"
```

Comment:
Instead of creating the resource immediately, we should instead generate a file with the secret YAML in it which then gets applied in `step_apply_resources`. This would only require adding the `--dry-run=client` and `-o yaml` options:
```
>> kubectl create secret docker-registry test-pull-secret -n astra --docker-username='<account_id>' --docker-password='<api_token>' --docker-server='<acs_registry>' --dry-run=client -o yaml

apiVersion: v1
data:
  .dockerconfigjson: eyJhdXRocyI6eyJcdTAwM2NhY3NfcmVnaXN0cnlcdTAwM2UiOnsidXNlcm5hbWUiOiJcdTAwM2NhY2NvdW50X2lkXHUwMDNlIiwicGFzc3dvcmQiOiJcdTAwM2NhcGlfdG9rZW5cdTAwM2UiLCJhdXRoIjoiUEdGalkyOTFiblJmYVdRK09qeGhjR2xmZEc5clpXNCsifX19
kind: Secret
metadata:
  creationTimestamp: null
  name: test-pull-secret
  namespace: astra
type: kubernetes.io/dockerconfigjson
```
You can then save this into a file in the operators folder. Tracking the file name will likely require adding a new 
`__GENERATED` variable alongside the others:
```
readonly __GENERATED_CRS_DIR="./astra-generated"
readonly __GENERATED_CRS_FILE="$__GENERATED_CRS_DIR/crs.yaml"
readonly __GENERATED_OPERATORS_DIR="$__GENERATED_CRS_DIR/operators"
readonly __GENERATED_KUSTOMIZATION_FILE="$__GENERATED_OPERATORS_DIR/kustomization.yaml"
readonly __GENERATED_PATCHES_TORC_FILE="$__GENERATED_CRS_DIR/post-deploy-patches_torc"
readonly __GENERATED_PATCHES_TRIDENT_OPERATOR_FILE="$__GENERATED_OPERATORS_DIR/post-deploy-patches_trident-operator"
# add  __GENERATED_TRIDENT_ACP_SECRET_FILE or something like that
```

Reply:
Done